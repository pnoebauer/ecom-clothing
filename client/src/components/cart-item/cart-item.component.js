import React from 'react';

import { 
	CartItemContainer, 
	ImageContainer, 
	ItemDetailsContainer
} from './cart-item.styles';

const CartItem = ({ item }) => {
	const {imageUrl, name, price, quantity} = item;
	return (
		<CartItemContainer>
			<ImageContainer src={imageUrl} alt='item' />
   			<ItemDetailsContainer>
				<span className='name'>{name}</span>
				<span className='price'>{quantity} x ${price}</span>
			</ItemDetailsContainer>
		</CartItemContainer>
	);
}
export default CartItem;