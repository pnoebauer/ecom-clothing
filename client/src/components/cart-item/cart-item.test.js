import React from 'react';
import {shallow} from 'enzyme';
import CartItem from './cart-item.component';
import {ItemDetailsContainer} from './cart-item.styles';

describe('CartItem component tests', () => {
	const mockItem = {
		imageUrl: 'www.testImage.com',
		price: 10,
		name: 'hats',
		quantity: 2,
	};
	const wrapper = shallow(<CartItem item={mockItem} />);

	test('render CartItem component', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('display correct info', () => {
		const {price, quantity, name} = mockItem;

		const priceText = wrapper.find('.price').text();
		expect(priceText).toEqual(`${quantity} x $${price}`);

		const nameText = wrapper.find('.name').text();
		expect(nameText).toEqual(`${name}`);
	});
});
