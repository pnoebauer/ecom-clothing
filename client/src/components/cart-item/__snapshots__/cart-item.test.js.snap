// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CartItem component 1`] = `
<cart-itemstyles__CartItemContainer>
  <cart-itemstyles__ImageContainer
    alt="item"
    src="www.testImage.com"
  />
  <cart-itemstyles__ItemDetailsContainer>
    <span
      className="name"
    >
      hats
    </span>
    <span
      className="price"
    >
      2
       x $
      10
    </span>
  </cart-itemstyles__ItemDetailsContainer>
</cart-itemstyles__CartItemContainer>
`;
